apps/rush-lib/src/logic/pnpm/PnpmShrinkwrapFile.ts

```diff
@@ -272,6 +272,12 @@ export class PnpmShrinkwrapFile extends BaseShrinkwrapFile {
     return packageDescription;
   }
 
+  private _getValidDependencyVersion(dependencyVersion: string): string {
+    return semver.valid(dependencyVersion.split('/').pop()!) ?
+          dependencyVersion.split('/').pop()! :
```

Comment:
Extract `dependencyVersion.split('/').pop()!` into a variable so that it need not be computed twice?